File: from-2501-to-2600/2559-count-vowel-strings-in-ranges/solution.go
func vowelStrings(words []string, queries [][]int) []int {
    prefix := make([]int, len(words) + 1)
    for i, word := range words {
        prefix[i + 1] = prefix[i]
        if isVowel(word[0]) && isVowel(word[len(word) - 1]){
            prefix[i + 1] += 1
        }
    }

    result := make([]int, len(queries))
    for i, query := range queries {
        left, right := query[0], query[1]
        result[i] = prefix[right + 1] - prefix[left]
    }
    return result
}

func isVowel(c byte) bool {
    if c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' {
        return true
    }
    return false
}

File: from-2501-to-2600/2559-count-vowel-strings-in-ranges/solution.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        n = len(words)
        prefix = [0] * (n + 1)

        for i in range(n):
            prefix[i + 1] = prefix[i]
            if words[i][0] in vowels and words[i][-1] in vowels:
                prefix[i + 1] += 1

        result = [0] * len(queries)
        for i, (left, right) in enumerate(queries):
            result[i] = prefix[right + 1] - prefix[left]

        return result
